File: templates/Uwp/_composition/_shared/Feature.ShareTarget/Activation/wts.ItemNameActivationHandler.cs
using System;
using System.Threading.Tasks;
using Windows.ApplicationModel.Activation;
using Windows.UI.Xaml;
using Windows.UI.Xaml.Controls;
using Param_RootNamespace.Views;

namespace Param_RootNamespace.Activation
{
    internal class wts.ItemNameActivationHandler : ActivationHandler<ShareTargetActivatedEventArgs>
    {
        protected override async Task HandleInternalAsync(ShareTargetActivatedEventArgs args)
        {
            // Activation from ShareTarget opens the app as a new modal window which requires a new activation.
            var frame = new Frame();
            Window.Current.Content = frame;
            frame.Navigate(typeof(wts.ItemNamePage), args.ShareOperation);
            Window.Current.Activate();

            await Task.CompletedTask;
        }
    }
}

File: templates/Uwp/_composition/_shared/Feature.ShareTarget._VB/Activation/wts.ItemNameActivationHandler.vb
Imports Param_RootNamespace.Views

Namespace Activation
    Friend Class wts.ItemNameActivationHandler
        Inherits ActivationHandler(Of ShareTargetActivatedEventArgs)

        Protected Overrides Async Function HandleInternalAsync(args As ShareTargetActivatedEventArgs) As Task
            ' Activation from ShareTarget opens the app as a new modal window which requires a new activation.
            Dim frame = New Frame()
            Window.Current.Content = frame
            frame.Navigate(GetType(wts.ItemNamePage), args.ShareOperation)
            Window.Current.Activate()

            Await Task.CompletedTask
        End Function
    End Class
End Namespace
